### ISSUE TITLE
False positive E0107.

### ISSUE THREAD

**Issue by JohnScience** [rust-lang/rust-analyzer#15821]
**rust-analyzer version**: 0.3.1713-standalone (f49320774 2023-10-28)

**rustc version**: doesn't work on both `rustc 1.72.1 (d5c2e9c34 2023-09-13)` and `rustc 1.74.0-nightly (ca62d2c44 2023-09-30)`

**relevant settings**: N/A

I have a generic struct [`CTOption`](https://github.com/JohnScience/ctoption/blob/24e3ba9c48afd5e7800c798bba40ed8ee4b344d4/src/lib.rs#L168) with two types-aliases that represent its parametrizations: [`CTSome`](https://github.com/JohnScience/ctoption/blob/24e3ba9c48afd5e7800c798bba40ed8ee4b344d4/src/lib.rs#L174) and [`CTNone`](https://github.com/JohnScience/ctoption/blob/24e3ba9c48afd5e7800c798bba40ed8ee4b344d4/src/lib.rs#L176).

For `CTSome`, there's an implementation of `new` that accepts one parameter. For `CTNone`, there's an implementation of `new` that accepts zero parameters.

When I try to use `CTNone::<u32>::new()`, I get "expected 1 argument, found 0 rust-analyzer ([E0107](https://doc.rust-lang.org/stable/error_codes/E0107.html))" error. Upon closer inspection, E0107 is "An incorrect number of generic arguments was provided.", which is straight up wrong.

Commit: https://github.com/JohnScience/ctoption/tree/24e3ba9c48afd5e7800c798bba40ed8ee4b344d4

**Comment by HKalbasi** [comment]
This is related to the const generics, I think there were a similar issue open but I couldn't find it. As a workaround, you can inline `IS_SOME` in the `CTSome` definition:
```
pub type CTSome<T> = CTOption<T, true>;
pub type CTNone<T> = CTOption<T, false>;
```

**Comment by ChayimFriedman2** [comment]
I fail to reproduce.
